src/gui/XournalView.cpp

```diff
@@ -377,6 +384,19 @@ void XournalView::pageSelected(size_t page) {
     control->updatePageNumbers(currentPage, pdfPage);
 
     control->updateBackgroundSizeButton();
+
+    if (control->getSettings()->isEagerPageCleanup()) {
+        this->cleanupBufferCache();
+    }
+
+    // Load surrounding pages if they are not
+    const auto& [pagesLower, pagesUpper] = preloadPageBounds(page, this->viewPages.size());
+    g_assert(pagesLower <= pagesUpper);
+    for (size_t i = pagesLower; i < pagesUpper; i++) {
+        if (this->viewPages[i]->getBufferPixels() == 0) {
+            this->viewPages[i]->rerenderPage();
+        }
+    }
```

Comment:
Do also not do this here. Same reason as above. Instead put this into scroll handling.